docs/api-data-sanitization.md

```diff
@@ -122,5 +141,141 @@ Options:
      value.
 
 
-
-
+### Response Schema Specification
+
+A "response schema" is a "JSON Schema Filter" structure, specifying how response (which must be JSON)
+should be filtered. Using this, you can implement a "default deny" approach to sanitizing API
+fields in a manner that may be more convenient than using JSON paths with conditional negations (a
+redact transform with a JSON path that matches all but an explicit list of named fields is the other
+approach to implementing 'default deny' style rules).
+
+Our "JSON Schema Filter" implementation attempts to align to the [JSON Schema](https://json-schema.org/specification-links.html)
+specification, with some variation as it is intended for *filtering* rather than *validation*. But
+generally speaking, you should be able to copy the JSON Schema for an API endpoint from its
+[OpenAPI specification](https://swagger.io/specification/) as a starting point for the
+`responseSchema` value in your rule set. Similarly, there are tools that can generate JSON Schema
+from example JSON content, as well as from data models in various languages, that may be useful.
+
+See: https://json-schema.org/implementations.html#schema-generators
+
+If a `responseSchema` attribute is specified for an `endpoint`, the response content will be
+*filtered* (rather than validated) against that schema. Eg, fields NOT specified in the schema, or
+not of expected type, will be removed from the response.
+
+  - `type` - one of :
+      - `object` a JSON object
+      - `array` a JSON array
+      - `string` a JSON string
+      - `number` a JSON number, either integer or decimal.
+      - `integer` a JSON integer (not a decimal)
+      - `boolean` a JSON boolean
+  - `properties` - for `type == object`, a map of field names to schema to filter field's value
+     against (eg, another `JsonSchemaFilter` for the field itself)
+  - `items` - for `type == array`, a schema to filter each item in the array against (again, a `JsonSchemaFilter`)
+  - `format` - for `type == string`, a format to expect for the string value. As of v0.4.38, this is
+     not enforced by the proxy.
+  - `$ref` - a reference to a schema specified in the `definitions` property of the root schema.
+  - `definitions` - a map of schema names to schemas of type `JsonSchemaFilter`; only supported at
+     root schema of endpoint.
+
+Example:
+
+The following is for a User from the GitHub API. See: https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-a-user
```

Comment:
The example is actually from getting users from GraphAPI: https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#the-graphql-endpoint

And I'd change the phrase to emphasize that with this example we are enforcing schema definition to match what we expect, avoiding to have any other field you could do over GraphQL

Reply:
👍 thanks - yeah, I see that I skimmed past as `graphql` path template when I was looking at example.

we should think of how to deal with graphql endpoints more generally; I guess only solution is an `or` at the root of the schema? (Eg, from graphql we expect `!user` or `!commit` etc?) Workable, but perhaps there's a better way.